GCTC-NTGC/gc-digital-talent
# Feature - remove soft skills from required

Resolves #4951 

## Notes
- Removes soft skills (skills in Transferable category) from the **Required** and **Nice to have** missing skills boxes (see [image](https://user-images.githubusercontent.com/16977254/204652762-a280577a-9368-41dd-a131-940a6a09edb5.png)).
- Also, removes soft skills from the Need-to-have and Nice-to-have section from the Experience Form page (see [image](https://user-images.githubusercontent.com/16977254/204374459-1bc8b39d-d829-4054-b183-2e53737f1f62.png)).
- Lastly, updates the `isApplicationComplete` check to remove all soft skills. Soft skills are now not required to submit an application.

## Testing
- Go to [Browse Opportunities](http://localhost:8000/en/browse/pools) and apply to a job with transferable skills.
- After applying ensure a new box for transferable skills appears matching this [image](https://user-images.githubusercontent.com/16977254/204652762-a280577a-9368-41dd-a131-940a6a09edb5.png).
- Go to Experience form and ensure no transferable skills appears in need and nice to have sections (see [image](https://user-images.githubusercontent.com/16977254/204374459-1bc8b39d-d829-4054-b183-2e53737f1f62.png))
- Lastly, add all required skills for the job and complete the application. Ensure user can move to step 2 and submit an application.
